tests/Validator.test.js

```diff
@@ -0,0 +1,797 @@
+import Validator from '../lib/Validator';
+
+const SCHEMA = {
+  id: 'test',
+  name: 'test',
+  types: {
+    string: {
+      uri: 'xsd:string',
+    },
+    normalizedString: {
+      uri: 'xsd:normalizedString',
+    },
+    token: {
+      uri: 'xsd:token',
+    },
+    integer: {
+      uri: 'xsd:integer',
+    },
+    nonNegativeInteger: {
+      uri: 'xsd:nonNegativeInteger',
+    },
+    float: {
+      uri: 'xsd:float',
+    },
+    double: {
+      uri: 'xsd:double',
+    },
+    decimal: {
+      uri: 'xsd:decimal',
+    },
+    date: {
+      uri: 'xsd:date',
+    },
+    dateTime: {
+      uri: 'xsd:dateTime',
+    },
+    time: {
+      uri: 'xsd:time',
+    },
+  },
+  classes: {
+    Test: {
+      name: 'Test',
+      attributes: {
+        a_string: {
+          name: 'a_string',
+          range: 'string',
+        },
+        a_normalized_string: {
+          name: 'a_normalized_string',
+          range: 'normalizedString',
+        },
+        a_token: {
+          name: 'a_token',
+          range: 'token',
+        },
+        an_integer: {
+          name: 'an_integer',
+          range: 'integer',
+        },
+        a_non_negative_integer: {
+          name: 'a_non_negative_integer',
+          range: 'nonNegativeInteger',
+        },
+        a_float: {
+          name: 'a_float',
+          range: 'float',
+        },
+        a_double: {
+          name: 'a_double',
+          range: 'double',
+        },
+        a_decimal: {
+          name: 'a_decimal',
+          range: 'decimal',
+        },
+        an_enum: {
+          name: 'an_enum',
+          range: 'Numbers',
+        },
+        a_date: {
+          name: 'a_date',
+          range: 'date',
+        },
+        a_datetime: {
+          name: 'a_datetime',
+          range: 'dateTime',
+        },
+        a_time: {
+          name: 'a_time',
+          range: 'time',
+        },
+        required_string: {
+          name: 'required_string',
+          range: 'string',
+          required: true,
+        },
+        a_big_number: {
+          name: 'a_big_number',
+          range: 'integer',
+          minimum_value: 100,
+        },
+        a_small_number: {
+          name: 'a_small_number',
+          range: 'integer',
+          maximum_value: 10,
+        },
+        a_medium_number: {
+          name: 'a_medium_number',
+          range: 'integer',
+          minimum_value: 25,
+          maximum_value: 75,
+        },
+        during_vancouver_olympics: {
+          name: 'during_vancouver_olympics',
+          range: 'date',
+          minimum_value: '2010-02-12',
+          maximum_value: '2010-02-28',
+        },
+        not_the_future: {
+          name: 'not_the_future',
+          range: 'date',
+          maximum_value: '{today}',
```

Comment:
So I see this works here, but for some reason its not working on the "sample collection date" in the Cancogen Covid19 schema.json template (which has an older todo=... way of specifying the min/max. which will be phased out in favour of minimum_value and maximum_value when LinkML allows dates there.) 

As a clearer test, I tried adding maximum_value:{today} to another field, the "sample received date" field, in the slot_definition, and in class attribute, and even in slot_usage, but none worked.

Is it perhaps because range of those fields includes two separate sources - one being "date" and the other being a null value menu?